x/slashing/keeper.go

```diff
@@ -376,6 +408,100 @@ func (k *Keeper) slashingSideDowntime(ctx sdk.Context, pack *SideDowntimeSlashPa
 	return nil
 }
 
+func (k *Keeper) slashingSideMalicousVote(ctx sdk.Context, pack *SideMalicousVoteSlashPackage) sdk.Error {
+	sideChainName, err := k.ScKeeper.GetDestChainName(pack.SideChainId)
+	if err != nil {
+		return ErrInvalidSideChainId(DefaultCodespace)
+	}
+	sideCtx, err := k.ScKeeper.PrepareCtxForSideChain(ctx, sideChainName)
+	if err != nil {
+		return ErrInvalidSideChainId(DefaultCodespace)
+	}
+
+	header := sideCtx.BlockHeader()
+	age := uint64(header.Time.Unix()) - pack.SideTimestamp
+	if age > uint64(k.MaxEvidenceAge(sideCtx).Seconds()) {
+		return ErrExpiredEvidence(DefaultCodespace)
+	}
+
+	validator := k.validatorSet.ValidatorByVoteAddr(sideCtx, pack.SideVoteAddr)
+	if validator == nil {
+		return ErrNoValidatorWithVoteAddr(k.Codespace)
+	}
+	sideConsAddr := []byte(validator.GetConsAddr())
+	if k.hasSlashRecord(sideCtx, sideConsAddr, MalicousVote, pack.SideHeight) {
+		return ErrDuplicateMalicousVoteClaim(k.Codespace)
+	}
+
+	slashAmt := k.MaliciousVoteSlashAmount(sideCtx)
+	validator, slashedAmt, err := k.validatorSet.SlashSideChain(ctx, sideChainName, sideConsAddr, sdk.NewDec(slashAmt))
+	if err != nil {
+		return ErrFailedToSlash(k.Codespace, err.Error())
+	}
+
+	maliciousVoteClaimFee := k.MaliciousVoteSlashFee(sideCtx)
```

Comment:
I think the fee is not needed here, we can simplify it.

Reply:
good idea